python/pyarrow/pandas-shim.pxi

```diff
@@ -58,6 +58,7 @@ cdef class _PandasAPIShim(object):
         self._array_like_types = (self._series, self._index,
                                   self._categorical_type)
 
+        self._version = pd.__version__
```

Comment:
review context: there is a `version` property that returns `self._version`, but this was never set